File: SB_Prototype_Pattern92/C++/SB_Prototype_Pattern92/Dog.cpp
#include "Dog.h"

Dog::Dog()
{
	std::wcout << L"Dog is Made" << std::endl;
}

Animal* Dog::makeCopy()
{

	std::wcout << L"Dog is Being Made" << std::endl;
	//Dog* dogObject = nullptr;

	// Calls the Animal super classes clone()
		// Then casts the results to Dog
		//dogObject = static_cast<Dog*>(__super::clone());
	Animal* dogObject = new Dog();

	return dogObject;
}

std::wstring Dog::toString()
{
	return L"Miss Violet is my Hero, Bark Growl Bark";
}

	

File: SB_Prototype_Pattern92/java/SB_Prototype_Java92/src/com/company/Dog.java
package com.company;

public class Dog implements Animal {

    public Dog(){
        System.out.println("Dog is Made");
    }

    public Animal makeCopy() {

        System.out.println("Dog is Being Made");
        Dog dogObject = null;

        try {
            // Calls the Animal super classes clone()
            // Then casts the results to Dog
            dogObject = (Dog) super.clone();
        }


        // If Animal didn't extend Cloneable this error
        // is thrown
        catch (CloneNotSupportedException e) {
            System.out.println("The Dog was Turned to Mush");
            e.printStackTrace();
        }
        return dogObject;
    }

    public String toString(){
        return "Miss Violet is my Hero, Bark Growl Bark";
    }


}
